Apps/W1/HybridSL/app/src/pages/CloudMigrationMgmtExt.PageExt.al

```diff
@@ -0,0 +1,230 @@
+            action(SLReRunHistoricalMigration)
+            {
+                ApplicationArea = All;
+                Caption = 'Rerun SL Historical Snapshot';
+                Enabled = HasCompletedSetupWizard;
+                Image = Archive;
+                ToolTip = 'Rerun SL Historical Snapshot';
+
+                trigger OnAction()
+                var
+                    SLCompanyAdditionalSettings: Record "SL Company Additional Settings";
+                    SLWizardIntegration: Codeunit "SL Wizard Integration";
+                begin
+                    if not SLCompanyAdditionalSettings.GetMigrateHistory() then begin
+                        Message(DetailSnapshotNotConfiguredMsg);
+                        exit;
+                    end;
+
+                    if Confirm(ConfirmRerunQst) then
+                        SLWizardIntegration.ScheduleSLHistoricalSnapshotMigration();
+                end;
+            }
+        }
+    }
+
+    trigger OnOpenPage()
+    var
+        HybridCompany: Record "Hybrid Company";
+        SLConfiguration: Record "SL Migration Config";
+        SLCompanyAdditionalSettings: Record "SL Company Additional Settings";
+    begin
+        HybridCompany.SetRange(Replicate, true);
+        HasCompletedSetupWizard := not HybridCompany.IsEmpty();
+
+
+        if HybridCompany.Get(CompanyName()) then begin
+            SLConfiguration.GetSingleInstance();
+            if GetHasCompletedMigration() then
+                if SLCompanyAdditionalSettings.GetMigrateHistory() then
+                    if not SLConfiguration.HasHistoricalJobRan() then
+                        ShowSLHistoricalJobNeedsToRunNotification();
+        end;
+    end;
+
+    internal procedure GetHasCompletedMigration(): Boolean
+    var
+        DataMigrationStatus: Record "Data Migration Status";
+    begin
+        DataMigrationStatus.SetFilter(Status, '%1|%2', DataMigrationStatus.Status::Completed, DataMigrationStatus.Status::"Completed with Errors");
+        exit(not DataMigrationStatus.IsEmpty());
+    end;
+
+    internal procedure ShowSLHistoricalJobNeedsToRunNotification()
+    var
+        SLHistoricalSnapshotJobHasNotRanNotification: Notification;
+    begin
+        SLHistoricalSnapshotJobHasNotRanNotification.Id := '253A52DA-F9F8-4B2E-B3C2-2E42DD8B97D5';
+        SLHistoricalSnapshotJobHasNotRanNotification.Recall();
+        SLHistoricalSnapshotJobHasNotRanNotification.Message := HistoricalDataJobNotRanMsg;
+        SLHistoricalSnapshotJobHasNotRanNotification.Scope := NotificationScope::LocalScope;
+        SLHistoricalSnapshotJobHasNotRanNotification.AddAction(HistoricalDataStartJobMsg, Codeunit::"SL Wizard Integration", 'StartSLHistoricalJobMigrationAction');
+        SLHistoricalSnapshotJobHasNotRanNotification.Send();
+    end;
+
+    trigger OnAfterGetRecord()
+    var
+        SLMigrationErrorOverview: Record "SL Migration Error Overview";
+    begin
+        MigrationErrorCount := SLMigrationErrorOverview.Count();
```

Comment:
we must not run this code nor the code in the trigger below if the cloud migration is not enabled.

Reply:
Code has been added to control visibility based on whether or not the SL Migration is enabled.